File: src/main/java/com/ytg123/manhunt/command/ClearCacheCommand.java
package com.ytg123.manhunt.command;

import com.mojang.brigadier.CommandDispatcher;
import com.ytg123.manhunt.ManhuntUtils;
import com.ytg123.manhunt.init.ManhuntPackets;
import io.netty.buffer.Unpooled;
import net.fabricmc.fabric.api.network.ServerSidePacketRegistry;
import net.minecraft.network.PacketByteBuf;
import net.minecraft.server.command.ServerCommandSource;

import static net.minecraft.server.command.CommandManager.*;

public class ClearCacheCommand {
    public static void register(CommandDispatcher<ServerCommandSource> dispatcher) {
        dispatcher.register(
                literal("clearManhuntCache").executes(context -> {
                    ManhuntUtils.haveMod.clear();
                    context.getSource().getMinecraftServer().getPlayerManager().getPlayerList().forEach(player -> {
                        ServerSidePacketRegistry.INSTANCE.sendToPlayer(player, ManhuntPackets.SERVER_QUESTION_PACKET_ID, new PacketByteBuf(Unpooled.buffer()));
                    });
                    return 1;
                })
        );
    }
}

File: manhunt-base/src/main/kotlin/io/github/ytg1234/manhunt/command/ClearCacheCommand.kt
package io.github.ytg1234.manhunt.command

import com.mojang.brigadier.Command
import com.mojang.brigadier.builder.LiteralArgumentBuilder
import io.github.ytg1234.manhunt.base.SERVER_QUESTION_PACKET_ID
import io.github.ytg1234.manhunt.base.haveMod
import io.github.ytg1234.manhunt.util.PermedCommand
import net.minecraft.server.command.CommandManager
import net.minecraft.server.command.ServerCommandSource
import org.quiltmc.qsl.networking.api.PacketByteBufs
import org.quiltmc.qsl.networking.api.ServerPlayNetworking

/**
 * Used to clear the [haveMod] list and resend the packet.
 *
 * @author YTG1234
 */
object ClearCacheCommand : PermedCommand("manhunt.command.clearcache", 2) {
    override val cmd: LiteralArgumentBuilder<ServerCommandSource> = CommandManager.literal("clearManhuntCache").executes { ctx ->
        haveMod.clear()
        ctx.source.server.playerManager.playerList.forEach {
            ServerPlayNetworking.send(it, SERVER_QUESTION_PACKET_ID, PacketByteBufs.empty())
        }
        Command.SINGLE_SUCCESS
    }
}
